### ISSUE TITLE
Use mailhog on mixed environment server?

### ISSUE THREAD

**Issue by paul-uz** [mailhog/MailHog#333]
Our server runs both the production and staging environments for a website.

I want to use mailhog, but only for the staging site.

If there an easy way to confgure mailhog to only catch emails sent from the staging site and not the live one?

**Comment by C-Duv** [comment]
Sorry if I'm a little direct here but:

Make the staging site send e-mails to Malhog and make the live site **not** send e-mails to Mailhog?

**Comment by paul-uz** [comment]
@C-Duv that's what I'm asking how to do 😄 

When I've used Mailhog in the past, it's just been on a VM and I have not had to configure the application specifically to send emails to Mailhog, it's just worked out of the box.

**Comment by C-Duv** [comment]
This has nothing to do with MailHog then. It's just a question of how you configure your applications/websites.

In the configuration of your staging website you have the IP/FQDN + port of the MailHog server as outgoing (SMTP) server.

So in the configuration of your live website use the IP/FQDN + port of a "real" outgoing (SMTP) server (GMail, OVH, etc. whatever it is...).

**Comment by paul-uz** [comment]
@C-Duv what aout if you are just using PHP's `mail()` function, and no specfic SMTP config?

**Comment by C-Duv** [comment]
Then you can't: because PHP `mail()` uses `sendmail` and I believe `sendmail` does not support multiple "outputs".

Maybe the `$additional_parameters` parameter of PHP `mail()`?

Or maybe MailHog could support an e-mail header that PHP would send (via `$additional_headers` parameter of `mail()`) and would tell MailHog to not catch and send it to the Internet.

But in the end it's not MailHog's job to do that... It's an SMTP server without Internet output (unless manually asked to) and an integrated webmail. It's goal is, to my point of view, to help developers during development phases without risking to send e-mails out in the Internet.

**Comment by paul-uz** [comment]
@C-Duv cool, I get ya 👍 

To be fair, we should be using SMTP, but thought I'd ask about PHP `mail()` incase it was possible.

**Comment by pimjansen** [comment]
Indeed, nothing todo with Mailhog. Sendmail will be used for php and will relay to localhost and maybe therefor Mailhog will pick it up.

tip: stop using native mail and implement for example the symfony mailer. Easy lib and easy to configure. Even as an one on one replacement.
